scripts/alert_emails/send_emails.ts

```diff
@@ -78,42 +82,55 @@ function generateSendData(usersToEmail: string[], alertForLocation: Alert) : Ema
     // TODO: maybe oauth here and add secrets into github.
     const auth = { apiKey: process.env.CREATE_SEND_TOKEN };
     const api: any = new createsend(auth);
-    const err_results: CampaignMonitorError[] = []
+    let emailSent = 0;
+    const uniqueEmailAddress: {[email: string]: number} = {}
+    const locationsWithEmails: {[fips: string]: number} = {}
 
-    const db = getFirebase().firestore();
+    const db = getFirestore();
     Object.keys(locationsWithAlerts).map(async (fips) => {
         db.collection(`snapshots/${currentSnapshot}/locations/${fips}/emails/`)
             .where("sentAt", "==", null).get().then(querySnapshot => {
                 querySnapshot.forEach(async doc => {
-                    await sendEmail(api, generateSendData([doc.id], locationsWithAlerts[fips]))
+                    await sendEmail(api, generateSendData([doc.id], locationsWithAlerts[fips]), dryRun)
                         .then(async result => {
-                            console.log(result)
-                            await db.collection(`snapshots/${currentSnapshot}/locations/${fips}/emails/`).doc(doc.id).set({sentAt: firebase.firestore.FieldValue.serverTimestamp()})
+                            emailSent += 1;
+                            uniqueEmailAddress[doc.id] = (uniqueEmailAddress[doc.id] || 0) + 1;
+                            locationsWithEmails[fips] = (locationsWithEmails[fips] || 0) + 1;
+                            if (dryRun) return;
+                            await db.collection(`snapshots/${currentSnapshot}/locations/${fips}/emails/`).doc(doc.id)
+                            .set({
+                                sentAt: admin.firestore.FieldValue.serverTimestamp()
+                            })
                         }).catch(err => {
                             console.log(err);
-                            err_results.push(err) // more elegantly handle the errors here?
+                            process.exit(-1);
                         });
                 })
             });
         });
-
+    console.log(`Total Emails to be sent: ${emailSent}. Total locations with emails ${locationsWithEmails}. Unique Email addresses ${Object(locationsWithEmails).keys.length}`)
```

Comment:
FYI- We'll need to update the info/alerts document in Firestore with the snapshot number once we're done so that the next time we send alerts it knows what to compare against.

Basically the counterpart to this code: https://github.com/covid-projections/covid-projections/pull/948/files#diff-a4941f1567c05d528ba48cd237b080c9R91

Reply:
okay added!